### DISCUSSION TITLE
Using Octane --watch with Vite HMR

### DISCUSSION THREAD

**Discussion by munezaclovis** [laravel/framework#51937] [Q&A]
Hello everybody

I have a problem.

I am using Livewire, Octane with the --watch option to reload my server when I make new changes and I also have Vite HMR.

Whenever I save changes, most of the times I get back an empty html page. these are the contents of the page:
![image](https://github.com/laravel/framework/assets/51137458/31b53475-c994-4882-8b68-24a771c48e8c)

My guess is that vite is fast that it reloads the browser before Octane finishes restarting the server therefore getting an empty page.

So far, I tried adding a delay of 2 seconds to the vite config but it just sucks sometimes.

Does anybody else have this issue? Maybe suggestions to fixing this? Am i doing something wrong?

THANK YOU!

**Comment by robsontenorio** [comment]
@munezaclovis any new workaround?

**Reply by munezaclovis** [reply]
I tried writing a custom plugin to hook into Vite so I could run `php artisan octane:reload` whenever there was a change (kinda replace the `--watch` on the artisan command that starts octane).
This still sucked since it resulted in a race condition almost everytime, so I ended up disabling Octane in local development (though I still use it in staging and production).
I use Frankenphp, so I didn’t have to change my Docker image to disable worker mode (Octane), it's just one command.

Plus, I made sure to set the `checkOctaneCompatibility: true` rule in phpstan config to catch anything I might miss.

**Comment by robsontenorio** [comment]
Found a solution while developing locally. 

Don’t use “—watch” and use  “—max-requests=1”

So, Octane will not put the code in cache. And Vite does its jobs refreshing the page when code changes.
